elf_reader.go

```diff
@@ -393,19 +395,51 @@ func (ec *elfCode) loadFunctions(section *elfSection) (map[string]asm.Instructio
 			// When splitting sections into subprograms, the targets of these calls
 			// are no longer in scope, so they must be resolved here.
 			if ins.IsFunctionReference() && ins.Constant != -1 {
-				tgt := jumpTarget(offset, ins)
+				tgt := jumpTarget(secOffset, ins)
 				sym := section.symbols[tgt].Name
 				if sym == "" {
-					return nil, fmt.Errorf("offset %d: no jump target found at offset %d", offset, tgt)
+					return nil, fmt.Errorf("offset %d: no jump target found at offset %d", secOffset, tgt)
 				}
 
-				ins.Reference = sym
+				ins.SetReference(sym)
 				ins.Constant = -1
 			}
 		}
 
+		// Only rebuild lineInfos at the start of a function
+		if funcOffset == 0 {
```

Comment:
This block should probably be moved to around line 381 above where the function body is rolled over.

I would make `ec.btf != nil` the enclosing check to avoid reallocating `lineInfos` for each function body, even if there is no BTF.

Reply:
Right, didn't realize we also enter that block on the very first iteration as well